hivemq4/k8s-image/Dockerfile

```diff
@@ -0,0 +1,50 @@
+ARG BASEIMAGE=hivemq/hivemq4:dns-latest
+ARG PROMETHEUS_EXTENSION_VERSION=4.0.4
+
+FROM busybox:1.35.0 as BUILD
+ARG PROMETHEUS_EXTENSION_VERSION
+
+WORKDIR /opt/hivemq/extensions
+#Next major version should not be compatible here
+COPY hivemq-k8s-sync-extension-4.?.?.zip ./hivemq-k8s-sync-extension.zip
+
+RUN unzip hivemq-k8s-sync-extension.zip -d /opt/hivemq/extensions/hivemq-k8s-sync-extension \
+    && rm hivemq-k8s-sync-extension.zip
+
+ADD https://www.hivemq.com/releases/extensions/hivemq-prometheus-extension-"${PROMETHEUS_EXTENSION_VERSION}".zip ./hivemq-prometheus-extension.zip
+RUN unzip hivemq-prometheus-extension.zip -d /opt/hivemq/extensions/hivemq-prometheus-extension \
+    && rm hivemq-prometheus-extension.zip
+COPY prometheusConfiguration.properties /opt/hivemq/extensions/hivemq-prometheus-extension/
+
+## Clean image to distribute
+FROM ${BASEIMAGE}
+
+ENV HIVEMQ_LOG_LEVEL=INFO\
+    HIVEMQ_CLUSTER_TRANSPORT_TYPE=TCP
+
+COPY config.xml /opt/hivemq/conf/config.xml
+COPY scripts/*.sh /opt/hivemq/bin/
+COPY --from=BUILD /opt/hivemq/extensions/* /opt/hivemq/extensions/
+
+### Move the entrypoint into the correct position & adjust for whether the base image uses entrypoint.d or not
+### Fix permissions as --chmod on COPY is not really working at the moment and requires buildkit
+RUN rm /opt/hivemq/extensions/prometheusConfiguration.properties \
+    && chmod +x /opt/hivemq/bin/*.sh \
```

Comment:
this line is unnecessary, the permission is already set in the base image

Reply:
Permissions are not maintained between multi-stage images https://github.com/moby/moby/issues/37123